src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -38,32 +56,53 @@ public ShooterSubsystem () {
      * 
      * @param RPM
      */
-    public void setFlywheelSpeed (double RPM) {
-        flywheelMotor.set(ControlMode.Velocity, UnitConversions.Shooter.flywheelEncoderTicksPer100msToRPM(RPM));
+    public void setFlywheelSpeeds (double mainRPM, double outerRPM) {
+        mainFlywheelMaster.set(ControlMode.Velocity, UnitConversions.Shooter.mainFlywheelEncoderTicksPer100msToRPM(mainRPM));
+        outerFlywheelMaster.set(ControlMode.Velocity, UnitConversions.Shooter.outerFlywheelEncoderTicksPer100msToRPM(outerRPM));
+    }
+
+    /**
+     * Set the flywheels to shooting speed (as defined in Constants.java)
+     */
+    public void activateFlywheels () {
+        setFlywheelSpeeds(Constants.Shooter.MAIN_FLYWHEEL_SHOOTING_RPM, Constants.Shooter.OUTER_FLYWHEEL_SHOOTING_RPM);
+    }
+
+    public double getMainFlywheelSpeed () {
+        return UnitConversions.Shooter.mainFlywheelEncoderTicksPer100msToRPM(mainFlywheelMaster.getSelectedSensorVelocity());
     }
 
-    public double getFlywheelSpeed () {
-        return UnitConversions.Shooter.flywheelEncoderTicksPer100msToRPM(flywheelMotor.getSelectedSensorVelocity());
+    public double getOuterFlywheelSpeed () {
+        return UnitConversions.Shooter.outerFlywheelEncoderTicksPer100msToRPM(outerFlywheelMaster.getSelectedSensorVelocity());
     }
 
     /**
      * @return if the closed loop error and flywheel speed rate of change
      * are below their tuned values
      */
     public boolean isFlywheelUpToSpeed () {
-        return (flywheelMotor.getClosedLoopError() <= Constants.Shooter.FLYWHEEL_SPEED_TOLERANCE) && 
-            (flywheelSpeedROC <= Constants.Shooter.FLYWHEEL_ROC_TOLERANCE);
+        boolean mainFlywheelUpToSpeed = (mainFlywheelMaster.getClosedLoopError() <= Constants.Shooter.MAIN_FLYWHEEL_SPEED_TOLERANCE) && 
+            (mainFlywheelSpeedROC <= Constants.Shooter.MAIN_FLYWHEEL_ROC_TOLERANCE);
+
+        boolean outerFlywheelUpToSpeed = (outerFlywheelSlave.getClosedLoopError() <= Constants.Shooter.OUTER_FLYWHEEL_SPEED_TOLERANCE) && 
+            (outerFlywheelSpeedROC <= Constants.Shooter.OUTER_FLYWHEEL_ROC_TOLERANCE);
+
+        return mainFlywheelUpToSpeed && outerFlywheelUpToSpeed;
     }
 
     public void stopFlywheels () {
-        flywheelMotor.stopMotor();
+        mainFlywheelMaster.stopMotor();
+        outerFlywheelMaster.stopMotor();
     }
 
     @Override
     public void periodic() {
         // Compute flywheel speed rate of change
-        double currentFlywheelSpeed = getFlywheelSpeed();
-        flywheelSpeedROC = (currentFlywheelSpeed - lastFlywheelSpeed) / t.get();
+        double mainFlywheelSpeed = getMainFlywheelSpeed();
```

Comment:
You might want to only calculate RoC at a set interval instead of every periodic. Periodic can change based on the CPU load, so your RoC could end up being inconsistent depending on how quickly the main loop is ticking.

In this case, you'd just early out if t.get() is below some threshold, so you only calculate every (e.g.) 200ms.

I'm not actually sure if this will be a problem in practice, it really depends on how busy the CPU is, but it's something to consider.

We do this in our game engine, instead of running as fast as the CPU can allow, we use a fixed update loop that only dispatches a system periodic on a set cadence to make sure that all updates are consistent.

Reply:
The robot code is already scheduled to run every 20ms, so this shouldn't be too much of an issue

Reply:
Turns out that 20ms is too fast, as the Talons update only every 100ms.